core/src/apps/common/keychain.py

```diff
@@ -54,13 +54,15 @@ def __init__(self, size: int) -> None:
         self.cache: dict[Any, Deletable] = {}
 
     def insert(self, key: Any, value: Deletable) -> None:
-        if key in self.cache_keys:
-            self.cache_keys.remove(key)
-        self.cache_keys.insert(0, key)
+        cache_keys = self.cache_keys  # local_cache_attribute
```

Comment:
missing simplification

Reply:
> it seems that there is missing simplification in a lot of places?

I am not seeing this, if simplification means "not including the object name in the cached name". The only places where it is done are for example `self_account_path` or `msg_safety_checks` where `account_path` or `safety_checks` are already defined symbols in the current scope